lib/analytics/reddit-analyzer.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+function analyze(redditJson) {
+  var i;
+  var stats = {
+    subreddit: {},
+    author: {},
+    domain: {},
+    ngram: {},
+    is_self: {},
+    gilded: 0,
+    num_comments: 0,
+    score: 0
+  };
+
+  if (redditJson.data && redditJson.data.children) {
+    for (i = 0; i < redditJson.data.children.length; i++) {
+      analyzeHelper(stats, "subreddit", redditJson.data.children[i].data);
+      analyzeHelper(stats, "author", redditJson.data.children[i].data);
+      analyzeHelper(stats, "domain", redditJson.data.children[i].data);
+      analyzeHelper(stats, "is_self", redditJson.data.children[i].data);
+
+      stats.gilded += redditJson.data.children[i].data.gilded;
+      stats.num_comments += redditJson.data.children[i].data.num_comments;
+      stats.score += redditJson.data.children[i].data.score;
+    }
+  }
+
+  console.log(stats);
+
+  return stats;
+};
+
+function analyzeHelper(stats, group, data) {
+  if (!stats[group][data[group]]) {
+    stats[group][data[group]] = {};
+  }
+
+  if (!stats[group][data[group]].count) {
+    stats[group][data[group]].count = 0;
+  }
```

Comment:
You could probably initialize an object with these keys at the start of the function, "count", "gilded" etc, instead of doing if not X -> set X to 0 -> set X to value.